lib/linkerd2-trace-context/src/propagation.rs

```diff
@@ -0,0 +1,248 @@
+use super::{Error, Flags, Id};
+use bytes::Bytes;
+use http::header::HeaderValue;
+use rand::{rngs::SmallRng, FromEntropy};
+use std::fmt;
+use tracing::{trace, warn};
+
+const HTTP_TRACE_ID_HEADER: &str = "x-b3-traceid";
+const HTTP_SPAN_ID_HEADER: &str = "x-b3-spanid";
+const HTTP_SAMPLED_HEADER: &str = "x-b3-sampled";
+
+const GRPC_TRACE_HEADER: &str = "grpc-trace-bin";
+const GRPC_TRACE_FIELD_TRACE_ID: u8 = 0;
+const GRPC_TRACE_FIELD_SPAN_ID: u8 = 1;
+const GRPC_TRACE_FIELD_TRACE_OPTIONS: u8 = 2;
+
+#[derive(Debug)]
+pub enum Propagation {
+    Http,
+    Grpc,
+}
+
+#[derive(Debug)]
+pub struct TraceContext {
+    pub propagation: Propagation,
+    pub version: Id,
```

Comment:
As far as I can tell, a `version` is always a single byte, right? In that case, IMO, it doesn't really make sense for it to be an `Id`, which is internally a `Vec`. Why not define a newtype like
```rust
#[derive(Default, Clone, ...)]
struct Version(u8);
```
instead? This would save us an allocation, and (if you take my advice above about separating the `Id` type into `TraceId`s and `SpanId`s) let us enforce those known length constraints at the type level rather than at runtime...

Reply:
Version was unused so I just removed it.